tests/unit/api/c/source/test_source.cpp

```diff
@@ -281,6 +694,19 @@ TEST_F(TestSource, SourceGetUniversesWorks)
   EXPECT_EQ(get_source_universes_fake.call_count, 1u);
 }
 
+TEST_F(TestSource, SourceGetUniversesHandlesNotFound)
+{
+  VERIFY_LOCKING(sacn_source_get_universes(kTestHandle, nullptr, 0u));
+  EXPECT_EQ(get_source_universes_fake.call_count, 0u);
+  SetUpSource(kTestHandle);
+  GetSource(kTestHandle)->terminating = true;
```

Comment:
Peeking into private state and manipulating it should always be a last resort when unit testing. Would it be the same test if you were to set the source to terminating by calling the API function to destroy it?